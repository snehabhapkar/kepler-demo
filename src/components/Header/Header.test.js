
import React from 'react';
import renderer from 'react-test-renderer';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';

import Header from './Header';

const anchorEl  = null;
const open = Boolean(anchorEl);


describe('Header', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <Header>
          <AppBar position="static" >
          <Toolbar>
            <Typography variant="h6" color="inherit" >
              GapMaps Admin
            </Typography>
              <div>
                <IconButton aria-owns={open ? 'menu-appbar' : undefined} aria-haspopup="true" onClick={jest.fn()} color="inherit">
                  <AccountCircle />
                </IconButton>
                <Menu id="menu-appbar" anchorEl={anchorEl}
                  anchorOrigin={{
                    vertical: 'top',
                    horizontal: 'right',
                  }}
                  transformOrigin={{
                    vertical: 'top',
                    horizontal: 'right',
                  }}
                  open={open}
                  onClose={jest.fn()}>
                  <MenuItem onClick={jest.fn()}>Logout</MenuItem>
                </Menu>
              </div>
          </Toolbar>
        </AppBar>
        </Header>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
