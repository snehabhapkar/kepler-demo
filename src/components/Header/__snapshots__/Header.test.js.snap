// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders children correctly 1`] = `
<div
  className="MenuAppBar-root-1"
>
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionFixed-6 MuiAppBar-colorPrimary-12 mui-fixed MenuAppBar-appBar-4"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <h6
        className="MuiTypography-root-45 MuiTypography-h6-62 MuiTypography-colorInherit-74 MenuAppBar-grow-2"
      >
        GapMaps Admin
      </h6>
      <div>
        <button
          aria-haspopup="true"
          className="MuiButtonBase-root-87 MuiIconButton-root-81 MuiIconButton-colorInherit-82"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-86"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-90"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-103"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
